src/main/java/com/shopify/minishopify/model/Role.java

```diff
@@ -0,0 +1,40 @@
+package com.shopify.minishopify.model;
+
+import javax.persistence.*;
+
+@Entity
+@Table(name = "roles")
+public class Role {
+
+    public enum ERole {
```

Comment:
From my understanding of this, this is hierarchy and each role includes all the privileges of the role beneath it. That would mean If someone was `ROLE_ADMIN` they would not need a `ROLE_USER` (similar to `ROLE_USER` vs `ROLE_ANON`).

Assuming that's correct, why would there need to be a Many-to-Many relationship between User and Role? Instead of a Many-to-One relationship between User and Role?

Reply:
While this is correct at the current state of the project, it's not unlikely that in a real scenario we would delegate roles that do not overlap in privelage. I can change the behavior to Many-to-one otherwise.